plone/supermodel/model.py

```diff
@@ -107,8 +107,27 @@ def finalizeSchemas(parent=Schema):
         )
 
     def walk(schema):
-        yield schema
-        for child in schema.dependents.keys():
+        if isinstance(schema, SchemaClass):
+            yield schema
+        else:
+            logger.exception(f"Got non-SchemaClass {schema}.")
```

Comment:
``logger.exception`` should only be called in a exception handler https://docs.python.org/3/library/logging.html#logging.Logger.exception